### PR TITLE
add translation for 'raise' BIF

### PR DESCRIPTION
**Pull request by the-mikedavis**
See [this](https://github.com/erlang/otp/blob/df48c260e74c3e9058ff8681ce9f554e6fa0fe34/lib/compiler/src/beam_asm.erl#L360-L361) line in beam_asm.

The `raise` instruction needs to be translated to a `bif`. Luckily
the arguments line up very well so it's an easy translation. It was
a bit tough writing a test case that the compiler wouldn't optimize
the raise/try out of, I ended up using the send+receive technique from
a few test cases down.

### CODE REVIEW

**Comment by michaelklishin** [comment]
Thank you for your ongoing contributions @the-mikedavis 👍 👍

File: src/khepri_tx.erl [hunk 1]
```diff
@@ -298,6 +300,12 @@ ensure_instruction_is_permitted({test_heap, _, _}) ->
     ok;
 ensure_instruction_is_permitted({trim, _, _}) ->
     ok;
+ensure_instruction_is_permitted({'try', _, _}) ->
+    ok;
+ensure_instruction_is_permitted({try_end, _}) ->
+    ok;
+ensure_instruction_is_permitted({try_case, _}) ->
+    ok;
```
Comment (dumbbell) [inline src/khepri_tx.erl:308]: Could you please split your commit into two and make a separate commit for this chunk?

**Review by dumbbell — CHANGES_REQUESTED** [review]
This looks good to me. I made a small comment to clarify the commit history.
